templates/extensions/the-graph/README.md.args.mjs

```diff
@@ -0,0 +1,247 @@
+
+> NOTE FOR LINUX USERS: If you are running Linux you will need some additional changes to the project.
+
+##### Linux Only
+
+Update your package.json in packages/hardhat with the following command line option for the hardhat chain.
+
+\`\`\`
+"chain": "hardhat node --network hardhat --no-deploy --hostname 0.0.0.0"
+\`\`\`
+
+Save the file and then restart your chain in its original window.
+
+\`\`\`
+yarn chain
+\`\`\`
+
+You might also need to add a firewall exception for port 8432. As an example for Ubuntu... run the following command.
+
+\`\`\`
+sudo ufw allow 8545/tcp
+\`\`\`
+
+&nbsp;
+
+#### ✅ Step 2: Create and ship our Subgraph ✅
+
+Now we can open up a fourth window to finish setting up The Graph. 😅 In this forth window we will create our local subgraph!
+
+> Note: You will only need to do this once.
+
+\`\`\`
+yarn local-create
+\`\`\`
+
+> You should see some output stating your Subgraph has been created along with a log output on your graph-node inside docker.
+
+Next we will ship our subgraph! You will need to give your subgraph a version after executing this command. (e.g. 0.0.1).
+
+\`\`\`
+yarn local-ship
+\`\`\`
+
+> This command does the following all in one… 🚀🚀🚀
+
+-   Copies the contracts ABI from the hardhat/deployments folder
+-   Generates the networks.json file
+-   Generates AssemblyScript types from the subgraph schema and the contract ABIs.
+-   Compiles and checks the mapping functions.
+-   … and deploy a local subgraph!
+
+> If you get an error ts-node you can install it with the following command
+
+\`\`\`
+npm install -g ts-node
+\`\`\`
+
+You should get a build completed output along with the address of your Subgraph endpoint.
+
+\`\`\`
+Build completed: QmYdGWsVSUYTd1dJnqn84kJkDggc2GD9RZWK5xLVEMB9iP
+
+Deployed to http://localhost:8000/subgraphs/name/scaffold-eth/your-contract/graphql
+
+Subgraph endpoints:
+Queries (HTTP):     http://localhost:8000/subgraphs/name/scaffold-eth/your-contract
+\`\`\`
+
+&nbsp;
+
+#### ✅ Step 3: Test your Subgraph ✅
+
+Go ahead and head over to your subgraph endpoint and take a look!
+
+> Here is an example query…
+
+\`\`\`
+  {
+    greetings(first: 25, orderBy: createdAt, orderDirection: desc) {
+      id
+      greeting
+      premium
+      value
+      createdAt
+      sender {
+        address
+        greetingCount
+      }
+    }
+  }
+\`\`\`
+
+> If all is well and you’ve sent a transaction to your smart contract then you will see a similar data output!
+
+#### ✅ Side Quest: Run a Matchstick Test ✅
+
+Matchstick is a [unit testing framework](https://thegraph.com/docs/en/developing/unit-testing-framework/), developed by [LimeChain](https://limechain.tech/), that enables subgraph developers to test their mapping logic in a sandboxed environment and deploy their subgraphs with confidence!
+
+The project comes with a pre-written test located in\`packages/subgraph/tests/asserts.test.ts\`
+
+To test simply type....
+
+\`\`\`
+yarn subgraph:test
```

Comment:
Getting 

Fetching latest version tag...
Downloading release from https://github.com/LimeChain/matchstick/releases/download/0.6.0/binary-linux-20
Error fetching release: Request failed with status code 404

on Ubuntu 20.04

Reply:
Might be related to https://github.com/LimeChain/matchstick/issues/418#issuecomment-1803767144, 

It suggest adding `test: graph test -d` (So that it runs this command in docker) could you try adding this in scaffolded app `packages/subgraph/package.json` and see if it works ? Lol just throwing a random stone not sure if it will work

Reply:
I was getting a message saying it's not available for Windows 10. Maybe we should add a warn in the Readme?

Reply:
> It suggest adding test: graph test -d (So that it runs this command in docker) could you try adding this in scaffolded app packages/subgraph/package.json and see if it works ? Lol just throwing a random stone not sure if it will work

It worked great for me on Windows!  TY

Reply:
Nicce, just added a note in readme in side quest section at https://github.com/scaffold-eth/create-eth/pull/25/commits/6272c812dd659c95fb677efcf435a8dc9f9cf54c, that you can run `yarn subgraph:test -d` if you get an error

Reply:
It worked with -d (and it installed a lot of things!)

```
Igniting tests 🔥

asserts
--------------------------------------------------
  Asserts:
    √ Greeting and Sender entities - 0.104ms

All 1 tests passed! 😎
```